client/dfclient/src/gameplay_state.cpp

```diff
@@ -432,6 +431,15 @@ GameplayState::GameplayState(Resources& r) : _resources(r), _deathReportProcesso
 }
 
 GameplayState::~GameplayState() {
+	// the order of these is important
```

Comment:
Maybe you could elaborate here how those should be ordered and why the order is important?